doc/adrs/160_add_support_for_styled_labels_in_view_model_for_trees.adoc

```diff
@@ -0,0 +1,179 @@
+= ADR-160 - Add support for styled labels in view model for trees
+
+== Context
+
+Currently, the tree description supports the ``StyledString`` in tree item label but no styles are given through the view model definition of trees.
+
+== Decision
+
+We need to update the tree view DSL to allow specifiers to express how labels should be stylized.
+We need also to enhance the way the italic and bold style is managed because it is suitable to be used in the view DSL.
+
+=== Tree view DSL
+
+Currently, the tree view DSL, ``TreeDescription`` only has the ``treeItemLabelExpression`` that returns the string to display regarding the given ``self`` element.
+We will replace this property by a new property to hold styled label descriptions.
+
+```
+TreeDescription {
+  ...
+  treeItemLabelDescriptions: TreeItemLabelDescription [0..*]
+  ...
+}
+
+TreeItemLabelDescription {
+  name: Identifier
+  preconditionExpression: InterpretedExpression<boolean>
+  children: TreeItemLabelElementDescription[1..*]
+}
+
+IfTreeItemLabelElementDescription extends TreeItemLabelElementDescription {
+  predicateExpression: InterpretedExpression<boolean>
+  children: TreeItemLabelElementDescription[1..*]
+}
+
+ForTreeItemLabelElementDescription extends TreeItemLabelElementDescription {
+  iterator: String
+  iterableExpression: InterpretedExpression<List<Object>>
+  children: TreeItemLabelElementDescription[1..*]
+}
+
+ReusedTreeItemLabelElementDescription extends TreeItemLabelElementDescription {
+  elements: TreeItemLabelElementDescription[0..*] (non containment reference)
+}
+
+TreeItemLabelFragmentDescription extends TreeItemLabelElementDescription {
+  labelExpression: String
+  style: TextStyleDescription [0..1]
```

Comment:
Please specifiy it is non containment reference

Reply:
Done